src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -47,6 +50,36 @@ public static unsafe bool TryGetHeaderName(IntPtr buffer, int length, out string
                 out name);
         }
 
+        public static string GetHeaderValue(string name, char[] array, int startIndex, int length)
+        {
+            Debug.Assert(name != null);
+            CharArrayHelpers.DebugAssertArrayInputs(array, startIndex, length);
+
+            if (length == 0)
+            {
+                return string.Empty;
+            }
+
+            // If it's a known header value, use the known value instead of allocating a new string.
+
+            // Do a really quick reference equals check to see if name is the same object as
+            // HttpKnownHeaderNames.ContentEncoding, in which case the value is very likely to
+            // be either "gzip" or "deflate".
+            if (ReferenceEquals(name, ContentEncoding))
+            {
+                if (CharArrayHelpers.EqualsOrdinalAsciiIgnoreCase(Gzip, array, startIndex, length))
+                {
+                    return Gzip;
```

Comment:
If the server sends "GZIP" on the wire, then this will result in a header collection that contains "gzip".  I realize these are semantically equivalent, but do we care about preserving the literal bytes from the wire?